torch/csrc/jit/script/compiler.cpp

```diff
@@ -473,6 +474,18 @@ at::optional<std::vector<Value*>> tryMatchSchema(
           value = graph.insertNode(graph.createUndefined())->output();
       }
 
+      //implicit conversion of tensors to scalars
+      if(convert_tensors_to_nums && arg.type->isSubtypeOf(NumberType::get())
+          && value->type()->isSubtypeOf(DynamicType::get())) {
+          auto err = checkScalarToNum(value->type(), arg.type);
```

Comment:
I don't like that fact that having a more concrete type here (like a sized tensor) increases the amount of error messages the compiler would emit. It could have some subtle behavior, such as moving on to another overload of this function, when the sizes are wrong but accepting this overload if it cannot determine the sizes. We should just allow this to fail.